docs/auth.md

```diff
@@ -0,0 +1,112 @@
+| 4   | grantRolesToUser         | Grants roles and privileges to a user. |
+| 5   | revokeRolesFromUser      | Removes roles from a user.             |
+| 6   | updateUser               | Updates a user's data.                 |
+| 7   | usersInfo                | Returns information about the users.   |
+
+## Commands that require authentication
+
+| #   | command          | Description                 |
+|-----|------------------|-----------------------------|
+| 1   | listCommands     | Returns a list of commands. |
+| 2   | connectionStatus | Returns connection status.  |
+
+
+# Tasks
+## Add support for separate connection pools for each user 
+
+*We will try to not store the username and password*.
+We should try to use PostreSQL's authentication mechanism.
+
+When a client connects with credentials specified in the connection string (username and password), authentication message would be sent to the server (`saslStart`).
+In order to create a new connection pool, we should use the username from the authentication message.
+
+When a client disconnects, we should close the connection pool.
+We should also close the system pool and all user pools when the server is shut down.
+
+To separate connection pools we should:
+* Use current connection pool as the `system` pool (we will use it later to query the database for user's credentials).
+* Handle `saslStart` message.
+* To distinguish between users we should add to the Handler instance a map of connection pools, where the key is the username.
+
+### Questions:
+* How to distinguish between users connected without authentication?
+* Should we use a different connection pool for each user (for example for each host+port pair)?
+
+### Tests:
+* Connect to the server with username and password in connection string twice.
+* Connect to the server with different usernames and passwords.
+
+## Add support for `createUser` command
+
+Add support for the `createUser` command with the following parameters:
+* createUser
+* pwd
+* mechanisms
+
+Not in the scope of this task:
+* customData
+* roles
+* writeConcern
+* authenticationRestrictions
+* digestPassword
+* comment
+
+User data should be stored in the PostgreSQL database (`admin` database?).
+Create a PostgreSQL user with provided credentials and auth mechanisms.
+Grant user access for all databases.
+
+### Tests:
+
+* Create user with username and password.
+* Create user with username and password and mechanisms.
+* Create user with username and password and mechanism that not supported or not exists.
+
+## Add flag `authenticationMechanisms`
+
+**Note:** This one should be discussed with the team.
+
+Add flag `authenticationMechanisms` to the FerretDB server.
+It should be an array of strings, containing the names of the supported authentication mechanisms.
+Possible values are:
+* SCRAM-SHA-1 (*not sure about this one as PostgreSQL doesn't support it*)
+* SCRAM-SHA-256
+* X509
+* GSSAPI (Kerberos)
+* PLAIN (LDAP SASL)
```

Comment:
As we agreed on call, not all of them will be used. 
Could you please add here "why" and the list of mechanics left?

Reply:
Moved it above and added comments.